### ISSUE TITLE
NgbCalendar.getNext() unexpected behaviour

### ISSUE THREAD

**Issue by ahdbk** [ng-bootstrap/ng-bootstrap#3146]
### Bug description:
getNext() is supposed to add a periode to a date.

for example if the date is  2019-04-26 when we add a year we expect to have 2020-04-26.

With the actual implementation we will have 2019-01-01

this is the actual method implementation: 

```
getNext(date: NgbDate, period: NgbPeriod = 'd', number = 1) {
    let jsDate = toJSDate(date);

    switch (period) {
      case 'y':
        return new NgbDate(date.year + number, 1, 1);
      case 'm':
        jsDate = new Date(date.year, date.month + number - 1, 1, 12);
        break;
      case 'd':
        jsDate.setDate(jsDate.getDate() + number);
        break;
      default:
        return date;
    }
    return fromJSDate(jsDate);
  }
```

**Comment by maxokorokov** [comment]
Thanks for reporting it, there is a PR opened for this since a long time → #1277
This should also fix keyboard navigation.

**Comment by maxokorokov** [comment]
Fixed by #3355, will be part of `5.1.2`
